core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -960,6 +965,32 @@ protected void waitUntilContainerStarted() {
         }
     }
 
+    private String checkForEmulation() {
+        try {
+            DockerClient dockerClient = DockerClientFactory.instance().client();
+            String imageId = getCurrentContainerInfo().getImageId();
```

Comment:
At this point, it is safe to use `containerInfo` and avoid performing a call to the server.